### PR TITLE
Enable super init not called check in pylint

### PR DESCRIPTION
**Pull request by zhongjiajie**
---
Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by zhongjiajie** [comment]
ref: https://github.com/apache/airflow/pull/7828#issuecomment-602414330

**Comment by potiuk** [comment]
Niiiiiiiice!

**Review by potiuk — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
Looks like CI is failing:

https://travis-ci.org/github/apache/airflow/jobs/665847017

**Comment by zhongjiajie** [comment]
Thiank to restart failing test. Will take a look at other exclude tomorrow, it's midnight in my timezone

File: airflow/utils/log/logging_mixin.py [hunk 1]
```diff
@@ -134,9 +136,6 @@ def __init__(self, stream):
         if 'stdout' in stream:
             self._use_stderr = False
 
-        # StreamHandler tries to set self.stream
-        Handler.__init__(self)  # pylint: disable=non-parent-init-called
```
Comment (mik-laj) [inline airflow/utils/log/logging_mixin.py]: Ohhh. There's a trick here. One init was intentionally omitted.

File: airflow/utils/log/logging_mixin.py [hunk 2]
```diff
@@ -126,6 +126,8 @@ class RedirectStdHandler(StreamHandler):
     sys.stderr/stdout at handler construction time.
     """
     def __init__(self, stream):
+        # StreamHandler tries to set self.stream
+        super().__init__(stream)
```
Comment (mik-laj) [inline airflow/utils/log/logging_mixin.py]: This is a change that causes a change in behavior.

File: airflow/utils/log/logging_mixin.py (hunk 1 above)
Comment (zhongjiajie) [reply airflow/utils/log/logging_mixin.py]: Yeah, I found out that too, but why?

File: airflow/utils/log/logging_mixin.py [hunk 3]
```diff
@@ -135,7 +135,7 @@ def __init__(self, stream):
             self._use_stderr = False
 
         # StreamHandler tries to set self.stream
-        Handler.__init__(self)  # pylint: disable=non-parent-init-called
+        super(Handler).__init__()  # pylint: disable=bad-super-call
```
Comment (zhongjiajie) [inline airflow/utils/log/logging_mixin.py]: And what do you think this change? @mik-laj

Comment (zhongjiajie) [reply airflow/utils/log/logging_mixin.py]: Seem that will also call parent init, maybe we should keep this line in original.

**Comment by zhongjiajie** [comment]
> Great. Wouldn't you like to review other exclusions too?
> These look easy to fix:
> deprecated-pragma,
> deprecated-string-function,
> deprecated-str-translate-call,
> deprecated-itertools-function,
> deprecated-types-field,
> deprecated-operator-function,
> deprecated-urllib-function,
> deprecated-sys-function,

I checkout and find all all by kaxil

**Comment by potiuk** [comment]
Looks good . Do we remove WIP from the name :) ?

**Comment by zhongjiajie** [comment]
Thanks everyone.

**Pull request merged**